src/hotspot/share/gc/shenandoah/shenandoahControlThread.cpp

```diff
@@ -125,6 +125,12 @@ void ShenandoahControlThread::run_service() {
     assert (!gc_requested || cause != GCCause::_last_gc_cause, "GC cause should be set");
 
     if (gc_requested) {
+      // Create the KlassInfoTable for Shenandoah only if JFR is enabled.
+      #if INCLUDE_JFR
```

Comment:
Here and elsewhere:
1. for single line pre-proc directives, consider the use of `JFR_ONLY()`.
2. For multi-line pre-proc directives, unindent all the way to the zeroth column, in keeping with hotspot style precedent, even though newer pre-processors accept indented directives.